packages/api/cms-api/src/page-tree/page-tree.module.ts

```diff
@@ -56,7 +56,19 @@ export class PageTreeModule {
         return {
             module: PageTreeModule,
             imports: [MikroOrmModule.forFeature([AttachedDocument, PageTreeNode, ...(Scope ? [Scope] : [])])],
-            providers: [PageTreeService, pageTreeResolver, repositoryProvider, pageTreeConfigProvider, PageExistsConstraint],
+            providers: [
+                PageTreeService,
+                pageTreeResolver,
+                repositoryProvider,
+                pageTreeConfigProvider,
+                {
+                    provide: PageExistsConstraint,
+                    useFactory: (pageTreeService: PageTreeService) => {
```

Comment:
Why do we need to use a factory now?

Reply:
For some reason I don't understand adding the PageTreeService dependency to ContentScopeGuard gave errors for PageExistsConstraint on nest startup. This somehow worked around it, although it is the exact same thing